code/datums/paygrades/paygrade.dm

```diff
@@ -54,8 +34,25 @@ GLOBAL_LIST_INIT(co_paygrades, list(
 	PAY_SHORT_MO4
 ))
 
-GLOBAL_LIST_INIT(wy_paygrades, list(
-	PAY_SHORT_WYC8,
-	PAY_SHORT_WYC9,
-	PAY_SHORT_WYC10
-))
+/datum/paygrade/New()
+	. = ..()
+	switch(default_faction)
+		if(FACTION_MARINE)
+			if(officer_grade)
+				GLOB.uscm_officer_paygrades += paygrade
+			if(officer_grade >= GRADE_FLAG)
+				GLOB.uscm_highcom_paygrades += paygrade
+		if(FACTION_WEYLAND,FACTION_PMC)
+			if(officer_grade >= GRADE_FLAG)
+				GLOB.wy_highcom_paygrades += paygrade
+
+/proc/setup_paygrades()
+	. = list()
+	for(var/I in subtypesof(/datum/paygrade))
+		var/datum/paygrade/PG = I
```

Comment:
it has been a minute since I touched dm but can't we just for(var/datum/paygrade/PG as anything in subtypesof(/datum/paygrade))

Reply:
I don't think you can because it's not actual datum instances but paths

Reply:
cthulu is correct, this is a valid pattern